### ISSUE TITLE
`.matchedGeometry` transition glitches on physical device

### ISSUE THREAD

**Issue by tarrouye** [nathantannar4/Transmission#68]
Here is a simple reproducible example:

```
import SwiftUI
import Transmission

struct HeroMoveExample: View {
  let colors: [Color] = [.blue, .red, .green, .orange, .pink, .purple, .gray]

  @State private var presentedColor: Color?

  var body: some View {
    NavigationStack {
      ScrollView(.vertical) {
        LazyVGrid(columns: [.init(.flexible()), .init(.flexible())]) {
          ForEach(colors, id: \.hashValue) { color in
            PresentationLink(
              transition: .matchedGeometry(options: .init(edges: .all, preferredCornerRadius: 10))
            ) {
              Destination(color: color)
                .onAppear {
                  presentedColor = color
                }
                .onDisappear {
                  presentedColor = nil
                }
            } label: {
              RoundedRectangle(cornerRadius: 10)
                .foregroundStyle(color)
                .frame(height: 100)
                // hide this cell when we are presenting it
                // to have a nicer effect where it looks like the
                // item leaves/fills the slot
                .opacity(presentedColor == color ? 0 : 1)
            }
          }
        }
      }
      .padding(.horizontal)
      .navigationTitle("Hero Move Example")
    }
  }
}

struct Destination: View {
  @Environment(\.presentationCoordinator) private var presentationCoordinator

  let color: Color

  var body: some View {
    TransitionReader { proxy in
      ZStack(alignment: .topTrailing) {
        color
          .ignoresSafeArea(.all)

        Image(systemName: "x.circle.fill")
          .resizable()
          .frame(width: 30, height: 30)
          .onTapGesture {
            presentationCoordinator.dismiss()
          }
          .padding(.horizontal)
          .opacity(proxy.progress) // fade with transition
      }
    }
  }
}

#Preview {
  HeroMoveExample()
}
```

Here's how it looks in the simulator, everything works fine and the corner radius applies properly:

https://github.com/user-attachments/assets/25a6e158-64f3-4bda-a5ac-6a39be4e72e7

And here's how it looks on my physical device (iPhone 14 Pro, iOS 18.1): 

https://github.com/user-attachments/assets/b3427699-3751-4735-a74a-e5a888ef7b03

As you can see, the animation looks pretty glitchy and the masking doesn't seem to apply properly.

**Comment by nathantannar4** [comment]
Looks like this was an iOS 18.1 regression. Fixed in `1.9.0`!

**Comment by tarrouye** [comment]
Thanks a bunch @nathantannar4 !!
